source/dofs/dof_handler_policy.cc

```diff
@@ -4428,6 +4428,13 @@ namespace internal
           Assert(level_ghost_owners.size() == neighbor_cell_list.size(),
                  ExcInternalError());
 
+
+          // before sending & receiving, make sure we protect this section with
+          // a mutex:
+          static Utilities::MPI::CollectiveMutex      mutex;
+          Utilities::MPI::CollectiveMutex::ScopedLock lock(
+            mutex, tria.get_communicator());
```

Comment:
So this section wasn't protected by a barrier already!...

Reply:
yep, none of these were. Somewhat scary, right? I think we got lucky that other operations around these did the synchronization for us.

Reply:
Yes. I really think we should audit the other places where we do Isends/Irecvs once you've got this one in...

Reply:
I did this already for this PR (I went through all MPI_ANY_SOURCE grep results).

Reply:
Ah, ok. I thought that there are other places in `dof_handler_policy.cc`, but maybe not. But we also don't have to do everything at once.